### PR TITLE
Fixes wallet-HUD interaction issues.

### PR DESCRIPTION
**Pull request by MarinaGryphon**
Newscasters and HUDs should now properly identify people who have wallets.

### CODE REVIEW

**Comment by Anewbe** [comment]
Tested and working? Does this work for PDAs and Communicators as well?

**Comment by MarinaGryphon** [comment]
If Get_ID works, it should. I'll test it in a moment.

File: code/game/machinery/newscaster.dm [hunk 1]
```diff
@@ -936,17 +936,11 @@ obj/item/weapon/newspaper/attackby(obj/item/weapon/W as obj, mob/user as mob)
...
-			if(istype(human_user.wear_id, /obj/item/device/pda))	//autorecognition, woo!
-				var/obj/item/device/pda/P = human_user.wear_id
-				if(P.id)
-					scanned_user = GetNameAndAssignmentFromId(P.id)
-				else
-					scanned_user = "Unknown"
-			else if(istype(human_user.wear_id, /obj/item/weapon/card/id))
-				var/obj/item/weapon/card/id/ID = human_user.wear_id
-				scanned_user = GetNameAndAssignmentFromId(ID)
+			var/obj/item/weapon/card/id/I = human_user.wear_id.GetID()
```
Comment (Leshana) [inline code/game/machinery/newscaster.dm]: I recommend changing this to simply `human_user.GetID()` so it will also support IDs held in active hand.

**Comment by MarinaGryphon** [comment]
Fixes #2051 and #4808 (wasn't sure which was still open, so I put both).

**Comment by MarinaGryphon** [comment]
Tested, works properly with PDAs, wallets, communicators, etc.

**Comment by TheFurryFeline** [comment]
Looks like the involved Issues are still open, the numbers being what Marina stated above.

**Pull request merged**